web/test/pages/roadmap.test.tsx

```diff
@@ -228,4 +239,132 @@ describe("roadmap page", () => {
+
+    const sectionStart = subject.getByTestId("section-start");
+    const sectionPlan = subject.getByTestId("section-plan");
+
+    expect(within(sectionStart).getByText("step2")).toBeVisible();
+    expect(within(sectionPlan).getByText("step1")).toBeVisible();
+  });
+
+  it("expands start section only", () => {
+    useMockRoadmap({
+      steps: [
+        generateStep({
+          name: "step1",
+          section: "PLAN",
+          tasks: [generateTask({ id: "task1" })],
+        }),
+        generateStep({
+          name: "step2",
+          section: "START",
+          tasks: [generateTask({ id: "task2" })],
+        }),
+      ],
+    });
+
+    useMockUserData({
+      preferences: generatePreferences({
+        roadmapOpenSections: ["START"],
+      }),
+    });
+
+    const subject = renderRoadmapPage();
+
+    const sectionStart = subject.getByTestId("section-start");
+    const sectionPlan = subject.getByTestId("section-plan");
+
+    expect(within(sectionStart).getByText("step2")).toBeVisible();
+    expect(within(sectionPlan).getByText("step1")).not.toBeVisible();
+  });
+
+  it("expands plan section only", () => {
+    useMockRoadmap({
+      steps: [
+        generateStep({
+          name: "step1",
+          section: "PLAN",
+          tasks: [generateTask({ id: "task1" })],
+        }),
+        generateStep({
+          name: "step2",
+          section: "START",
+          tasks: [generateTask({ id: "task2" })],
+        }),
+      ],
+    });
+
+    useMockUserData({
+      preferences: generatePreferences({
+        roadmapOpenSections: ["PLAN"],
+      }),
+    });
+
+    const subject = renderRoadmapPage();
+
+    const sectionStart = subject.getByTestId("section-start");
+    const sectionPlan = subject.getByTestId("section-plan");
+
+    expect(within(sectionStart).getByText("step2")).not.toBeVisible();
+    expect(within(sectionPlan).getByText("step1")).toBeVisible();
+  });
+
+  it("updates accordion state", async () => {
+    useMockRoadmap({
+      steps: [
+        generateStep({
+          name: "step1",
+          section: "PLAN",
+          tasks: [generateTask({ id: "task1" })],
+        }),
+        generateStep({
+          name: "step2",
+          section: "START",
+          tasks: [generateTask({ id: "task2" })],
+        }),
+      ],
+    });
+
+    const userData = generateUserData({
+      preferences: generatePreferences({
+        roadmapOpenSections: ["PLAN", "START"],
+      }),
+    });
+
+    setupStatefulUserDataContext();
+    const subject = render(
+      <WithStatefulUserData initialUserData={userData}>
+        <RoadmapPage displayContent={{ contentMd: "" }} />)
+      </WithStatefulUserData>
+    );
+
+    const sectionPlan = subject.container.querySelector("#plan-header");
+    expect(sectionPlan).toBeInTheDocument();
+    fireEvent.click(sectionPlan as Element);
+    expect(currentUserData().preferences.roadmapOpenSections).toEqual(["START"]);
+  });
```

Comment:
It seems like we're only testing the functionality in RoadmapBySections but not the logic in MiniRoadmap. Seems like we should have a similar set of tests in MiniRoadmap. 

That, or if you like the suggestion of refactoring the logic in RoadmapBySections and MiniRoadmap into a shared component that they both use, then you could potentially consolidate those tests into a single unit test for that new component.

Reply:
Again not to sound like a broken record (I'm sorry!!) but in a similar way as above, if I write the test first, I would not add any functionality to MiniRoadmap until I had a failing test to drive it out. There are many ways of writing tests and you don't have to do it the way I do, of course!!! But I am always excited to show the benefits

Reply:
Added tests for MiniRoadmap.